src/Queue/QueueHandler.php

```diff
@@ -60,78 +66,78 @@ public function __construct(
-    {
-        try {
-            $this->raiseBeforeJobEvent($connectionName, $job);
-
-            $job->fire();
-
-            $this->raiseAfterJobEvent($connectionName, $job);
-        } catch (Throwable $exception) {
-            $this->raiseExceptionOccurredJobEvent($connectionName, $job, $exception);
-
-            $this->exceptions->report($exception);
-
-            throw $exception;
         }
     }
 
     /**
-     * @see \Illuminate\Queue\Worker::raiseBeforeJobEvent()
+     * ...
+     * 
+     * @param  array  $message 
+     * @return array 
      */
-    protected function raiseBeforeJobEvent(string $connectionName, SqsJob $job): void
+    protected function normalizeMessage(array $message): array
     {
-        $this->container->make(LogManager::class)
-            ->info("Processing job {$job->getJobId()}", ['name' => $job->resolveName()]);
-
-        $this->events->dispatch(new JobProcessing($connectionName, $job));
+        return [
+            'MessageId' => $message['messageId'],
+            'ReceiptHandle' => $message['receiptHandle'],
+            'Body' => $message['body'],
+            'Attributes' => $message['attributes'],
+            'MessageAttributes' => $message['messageAttributes'],
+        ];
     }
 
     /**
-     * @see \Illuminate\Queue\Worker::raiseAfterJobEvent()
+     * Convert SQS message to Laravel SQS job.
+     * 
+     * @param  array  $message 
+     * @return \Illuminate\Queue\Jobs\SqsJob 
      */
-    protected function raiseAfterJobEvent(string $connectionName, SqsJob $job): void
+    protected function buildJob(array $message): SqsJob
     {
-        $this->container->make(LogManager::class)
-            ->info("Processed job {$job->getJobId()}", ['name' => $job->resolveName()]);
-
-        $this->events->dispatch(new JobProcessed($connectionName, $job));
+        return new SqsJob(
+            $this->container,
+            $this->sqs,
+            $message,
+            $this->connection,
+            $this->queue,
+        );
     }
 
     /**
-     * @see \Illuminate\Queue\Worker::raiseExceptionOccurredJobEvent()
+     * Gather all of the queue worker options as a single object.
+     * 
+     * @return \Illuminate\Queue\WorkerOptions 
      */
-    protected function raiseExceptionOccurredJobEvent(string $connectionName, SqsJob $job, Throwable $th): void
+    protected function gatherWorkerOptions(): WorkerOptions
     {
-        $this->container->make(LogManager::class)
-            ->error("Job failed {$job->getJobId()}", ['name' => $job->resolveName()]);
+        $options = [
+            0, // backoff
+            512, // memory
+            0, // timeout
+            0, // sleep
+            3, // maxTries
+            false, // force
+            false, // stopWhenEmpty
+            0, // maxJobs
+            0, // maxTime
+        ];
+
+        if (property_exists(WorkerOptions::class, 'name')) {
+            $options = array_merge(['default'], $options);
+        }
```

Comment:
@georgeboot: Should we make these configurable somehow?

Reply:
I believe they are all fake values. Laravels queue worker expects some options, but at least in Vapor, they are just fake values.